cpp/src/arrow/array/concatenate.cc

```diff
@@ -436,6 +438,23 @@ class ConcatenateImpl {
     return Status::OK();
   }
 
+  Status Visit(const RunEndEncodedType& type) {
```

Comment:
These changes seem to lack tests, don't they?

Reply:
Very likely. I didn't add extra tests from what Tobias' PRs already had. Adding tests is part of my review of this code, so soon to come.